### PR TITLE
[browser][MT] cleanup

### PR DESCRIPTION
**Pull request by pavelsavara**
- rename `set_arg_bool`, `get_arg_bool` and fix type to `setB32` and `getB32`
- simplified `browser-threads` sample
- fix `Debug` wrapper functions
- fix MT memory leak with `is_receiver_should_free`
- disable `AggressiveInlining` in `DEBUG`
- fix `size` for the `alloc_stack_frame`
- `set_arg_proxy_context` when marshaling `null` to C#
- renamed `isManagedRunningOnCurrentThread`
- added `runtimeHelpers.currentThreadTID`
- fix `__emscripten_thread_init` wrapper

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public static void CallEntrypoint(JSMarshalerArgument* arguments_buffer)
             }
             catch (Exception ex)
             {
-                Environment.FailFast("CallEntrypoint: Unexpected synchronous failure. " + ex);
+                Environment.FailFast($"CallEntrypoint: Unexpected synchronous failure in {Environment.CurrentManagedThreadId}: " + ex);
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs:48]:
```suggestion
                Environment.FailFast($"CallEntrypoint: Unexpected synchronous failure (ManagedThreadId {Environment.CurrentManagedThreadId}): " + ex);
```

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs [hunk 2]
```diff
@@ -110,7 +110,7 @@ public static void ReleaseJSOwnedObjectByGCHandle(JSMarshalerArgument* arguments
             }
             catch (Exception ex)
             {
-                Environment.FailFast("ReleaseJSOwnedObjectByGCHandle: Unexpected synchronous failure. " + ex);
+                Environment.FailFast($"ReleaseJSOwnedObjectByGCHandle: Unexpected synchronous failure in {Environment.CurrentManagedThreadId}: " + ex);
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs:113]:
```suggestion
                Environment.FailFast($"ReleaseJSOwnedObjectByGCHandle: Unexpected synchronous failure (ManagedThreadId {Environment.CurrentManagedThreadId}): " + ex);
```

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs [hunk 3]
```diff
@@ -205,7 +205,7 @@ public static void CompleteTask(JSMarshalerArgument* arguments_buffer)
             }
             catch (Exception ex)
             {
-                Environment.FailFast("CompleteTask: Unexpected synchronous failure. " + ex);
+                Environment.FailFast($"CompleteTask: Unexpected synchronous failure in {Environment.CurrentManagedThreadId}: " + ex);
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs:208]:
```suggestion
                Environment.FailFast($"CompleteTask: Unexpected synchronous failure (ManagedThreadId {Environment.CurrentManagedThreadId}): " + ex);
```

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs [hunk 4]
```diff
@@ -280,7 +280,7 @@ public static void BindAssemblyExports(JSMarshalerArgument* arguments_buffer)
             }
             catch (Exception ex)
             {
-                Environment.FailFast("BindAssemblyExports: Unexpected synchronous failure. " + ex);
+                Environment.FailFast($"BindAssemblyExports: Unexpected synchronous failure in {Environment.CurrentManagedThreadId}: " + ex);
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs:283]:
```suggestion
                Environment.FailFast($"BindAssemblyExports: Unexpected synchronous failure (ManagedThreadId {Environment.CurrentManagedThreadId}): " + ex);
```

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSMarshalerArgument.cs [hunk 5]
```diff
@@ -111,11 +115,11 @@ internal JSProxyContext ToManagedContext
                 // during JSExport, this is marshaling parameters and it would be set by:
                 //    - alloc_stack_frame
                 //    - set_js_handle/set_gc_handle
-                var proxyContextGCHandle = (GCHandle)slot.ContextHandle;
-                if (proxyContextGCHandle == default)
+                if (slot.ContextHandle == IntPtr.Zero)
                 {
                     Environment.FailFast($"ContextHandle not set, ManagedThreadId: {Environment.CurrentManagedThreadId}. {Environment.NewLine} {Environment.StackTrace}");
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSMarshalerArgument.cs:120]:
```suggestion
                    Environment.FailFast($"ContextHandle not set (ManagedThreadId {Environment.CurrentManagedThreadId}): {Environment.NewLine} {Environment.StackTrace}");
```

**Review by maraf — APPROVED** [review]

File: src/mono/browser/runtime/pthreads/worker-thread.ts [hunk 6]
```diff
@@ -82,10 +80,9 @@ export function mono_wasm_pthread_on_pthread_created(): void {
     if (!WasmEnableThreads) return;
     try {
         forceThreadMemoryViewRefresh();
-
         const pthread_id = mono_wasm_pthread_ptr();
-        mono_assert(!is_nullish(pthread_id), "pthread_self() returned null");
-        monoThreadInfo.pthreadId = pthread_id;
+        mono_assert(pthread_id == monoThreadInfo.pthreadId, "needs to match");
```
Comment (maraf) [inline src/mono/browser/runtime/pthreads/worker-thread.ts:84]:
```suggestion
        mono_assert(pthread_id == monoThreadInfo.pthreadId, `needs to match (mono_wasm_pthread_ptr ${pthread_id}, threadId from thread info ${monoThreadInfo.pthreadId})`);
```

File: src/mono/browser/runtime/pthreads/worker-thread.ts [hunk 7]
```diff
@@ -210,14 +207,20 @@ export function replaceEmscriptenPThreadWorker(modulePThread: PThreadLibrary): v
...
-    };
     modulePThread.threadInitTLS = (): void => {
         originalThreadInitTLS();
         mono_wasm_pthread_on_pthread_created();
     };
-}
\ No newline at end of file
+}
+
+export function replaceEmscriptenPThreadWorker2(): void {
```
Comment (ilonatommy) [inline src/mono/browser/runtime/pthreads/worker-thread.ts:217]: This is an interesting name. Why we have `replaceEmscriptenPThreadWorker` and `replaceEmscriptenPThreadWorker2`? Let's figure out something more descriptive

Comment (pavelsavara) [reply src/mono/browser/runtime/pthreads/worker-thread.ts:217]: ideas welcome

Comment (ilonatommy) [reply src/mono/browser/runtime/pthreads/worker-thread.ts:217]: `replaceEmscriptenPThreadWorker` -> `replaceEmscriptenPThreadWorkerFromTLS` 
`replaceEmscriptenPThreadWorker2` -> `replaceEmscriptenPThreadWorkerFromModule`

**Comment by pavelsavara** [comment]
created active issue https://github.com/dotnet/runtime/issues/98758

**Pull request merged**